aws-s3-transfer-manager/src/operation/upload_objects/worker.rs

```diff
@@ -617,4 +703,35 @@ mod unit {
             );
         }
     }
+
+    #[tokio::test]
```

Comment:
I think we should add a test that tries to do MPU but fails and calls abortMultipart instead of completeMultipart.

Reply:
Good point. As discussed offline, a good place to add that test is probably [an integration test file for uploading single object](https://github.com/awslabs/aws-s3-transfer-manager-rs/blob/main/aws-s3-transfer-manager/tests/upload_test.rs).